classes/page_group.php

```diff
@@ -34,6 +34,25 @@ class page_group extends persistent {
     /** Name of the cache used in this class. */
     public const CACHE_NAME = 'page_group_metadata';
 
+    /**
+     * Returns the current month in YYYYMM format.
+     * @return int
+     */
+    public static function get_current_month(): int {
+        return (int) userdate(time() - 360, '%Y%m');
+    }
+
+    /**
+     * Gets the number of records stored for a paricular month.
+     *
+     * @param int $month
+     * @return int
+     */
+    public static function count_records_for_month(int $month): int {
+        global $DB;
+        return $DB->count_records(self::TABLE, ['month' => $month]);
```

Comment:
Since the `count_records_for_month` function is only used to check for emptyness/existence for nexturl/prevurl in the `page_groups.php` file. Could we instead change the function to perform a `record_exists` check instead, which should be faster theoretically as it does not need to compute the sum.

Reply:
done